v1/plugins/rest/auth.go

```diff
@@ -153,6 +157,9 @@ func (ap *bearerAuthPlugin) NewClient(c Config) (*http.Client, error) {
 
 func (ap *bearerAuthPlugin) Prepare(req *http.Request) error {
 	token := ap.Token
+	if ap.logger == nil {
+		ap.logger = logging.Get()
```

Comment:
I think the logger should be provided. This `Get` call is deprecated. Just noting this if we have to remove it later.